src/main/java/com/coveo/saml/SamlClient.java

```diff
@@ -304,6 +315,67 @@ public void redirectToIdentityProvider(HttpServletResponse response, String rela
+    //- SAML processors should not use HTTP error codes instead SAML error coded need to be used.
+    //
+    //Following are the URL parameter names supported in this binding.
+    //- SAMLRequest — SAML request message
+    //- RelayState — RelayState value
+    //nb. these are additional to POST...
+    //- SAMLEncoding — indicate the SAML encoding mechanism
+    //- SigAlg — Signature algorithm identifier
+    //- Signature — Base64 encoded signature value
+    //
+    //Example: https://localhost:9443/samlsso?SAMLRequest=nZPBjpskDyneY4s...&SigAlg=http%3A%2F%2Fwww.w3.org%2F2000%2F09%2Fxmldsig%23rsa-sha1&Signature=VoTUhZQVI1y...
+    //
+    //From: https://www.oasis-open.org/committees/download.php/35387/sstc-saml-bindings-errata-2.0-wd-05-diff.pdf
+    //1. Any signature on the SAML protocol message, including the <ds:Signature> XML element itself,MUST be removed.
+    //   Note that if the content of the message includes another signature, such as asigned SAML assertion, this embedded signature is not removed.
+    //   However, the length of such amessage after encoding essentially precludes using this mechanism.
+    //   Thus SAML protocolmessages that contain signed content SHOULD NOT be encoded using this mechanism.
+    //2. The DEFLATE compression mechanism, as specified in [RFC1951] is then applied to the entireremaining XML content of the original SAML protocol message.
+    //3. The compressed data is subsequently base64-encoded according to the rules specified in  IETFRFC 2045 [RFC2045].
+    //   Linefeeds or other whitespace MUST be removed from the result.
+    //4. The base-64 encoded data is then URL-encoded, and added to the URL as a query stringparameter which MUST be named SAMLRequest (if the message is a SAML request) or
+    //   SAMLResponse (if the message is a SAML response).
+    //5. If RelayState data is to accompany the SAML protocol message, it MUST be URL-encoded and placed in an additional query string parameter named RelayState.
+    //6. If the original SAML protocol message was signed using an XML digital signature, a new signaturecovering the encoded data as specified above MUST be attached using the rules stated below.
+    //
+    //SAMLRequest=value&RelayState=value&SigAlg=value "Finally, note that if there is no RelayState value, the entire parameter should be omitted from thesignature computation (and not included as an empty parameter name)."
+    //
+    StringBuilder sb = new StringBuilder();
+    sb.append("SAMLRequest=").append(URLEncoder.encode(getSamlRequest(SamlClient.SamlIdpBinding.Redirect), "UTF-8"));
+
+    if (relayState != null)
+      sb.append("&RelayState=").append(URLEncoder.encode(relayState, "UTF-8"));
+
+    sb.append("&SigAlg=").append( "http%3A%2F%2Fwww.w3.org%2F2000%2F09%2Fxmldsig%23rsa-sha1" );
+
+    byte[] bytesToSign = sb.toString().getBytes("UTF-8");
+    try {
+      java.security.Signature sig = java.security.Signature.getInstance("SHA1withRSA");
```

Comment:
Is there a particular reason to use SHA1? From what I can tell SHA1 is no longer considered a safe hash to use with RSA.

I think most vendors recommend switching to at least SHA256, which is also the algorithm that is used in `SamlClient::signSAMLObject`.
If there are no objections to using SHA256, the switch could be made by replacing rsa-sha1 with rsa-sha256 in the URL above, and changing SHA1withRSA to SHA256withRSA in this line.

Reply:
(Just a comment from a random contributor, btw. I have no review powers)

Reply:
That makes sense, thanks.  Testing with SHA256 now, then I'll issue a new pull request.